File: 4.8-bc/go/stage-3-hexagonal/catalog/app/service.go
package app

import (
	"context"

	"github.com/genkovich/claude-course-demos/4.8-bc/go/stage-3-hexagonal/catalog/domain"
)

type Service struct {
	repo domain.Repository
}

func New(repo domain.Repository) *Service {
	return &Service{repo: repo}
}

func (s *Service) List(ctx context.Context) ([]domain.Product, error) {
	return s.repo.List(ctx)
}

File: 4.8-bc/ts/stage-3-hexagonal/src/catalog/app/service.ts
// Catalog use case — список продуктів.
import type { Product } from "../domain/product.js";
import type { Repository } from "../domain/repository.js";

export class Service {
  constructor(private readonly repo: Repository) {}

  list(): Promise<Product[]> {
    return this.repo.list();
  }
}
